src/sagemaker_containers/env.py

```diff
@@ -598,6 +598,38 @@ def framework_module(self):  # type: () -> str
                 my_module:main"""
         return self._framework_module
 
+    def write_new_file(self, dir_name, file_name, file_content=None):  # type: (str, str, str) -> None
+        """Create a new file with customized content.
+        Args:
+            dir_name: name of the directory
+            file_name: name of the file
+            file_content: content of file
+        """
+        file_path = os.path.join(dir_name, file_name)
+        if os.path.exists(file_path):
+            warning_msg = 'File {} already exists and will be overwritten! ' \
+                          'Please check if file with the same name is created before this call.' \
+                .format(file_path)
+            logger.warning(warning_msg)
+
+        with open(file_path, 'w') as f:
+            if file_content is not None:
```

Comment:
I don't want to fail it here since it's easy not to fail. And since operation is in a try-catch block with the actual training funciton. I don't want to raise exception here. That try-catch block only catching exception from train function would be ideal.

What if I default content to be empty string. Then no need to check and writing empty string is equivalent to not writing anything.

Reply:
We need to raise both errors but differentiate Client Errors from Framework errors.

I suggest that any ClientError should wrapped in an Error class that we provide, for example: https://github.com/mvsusp/sagemaker-containers/blob/mvs-sm/src/sagemaker_containers/modules.py#L143

Reply:
I removed the check here. I agree we should have more specific exception. But that sounds like a separate task. It would be too large just for this task.

If it's ok, I won't do it here. Instead, I created a task just for better exception?